### ISSUE TITLE
Binding IsSelected is not working in Treeview

### ISSUE THREAD

**Issue by italoaguiar** [microsoft/microsoft-ui-xaml#5738]
Binding IsSelected is not working in Treeview (SelectionMode="Multiple").

**Steps to reproduce the bug**
1 - Run the code below 
`<muxc:TreeView ItemsSource="{x:Bind ViewModel.Items}" SelectionMode="Multiple">`
`                        <muxc:TreeView.ItemTemplate>`
`                           <DataTemplate>`
`                                <muxc:TreeViewItem ItemsSource="{Binding Children}" IsSelected="{Binding Selected}" Content="{Binding Name}" />`
`                            </DataTemplate>`
`                        </muxc:TreeView.ItemTemplate>`
`                    </muxc:TreeView>`

**Expected behavior**
Select items where Selected=True 

**Version Info**

NuGet package version:  2.7.0-prerelease.210816001

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| Yes |  |

| Windows version                    | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (22000.132)              | Yes   |
| May 2021 Update (19043)            |    |
| October 2020 Update (19042)        |    |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Comment by seven-mile** [comment]
I encountered similar problem on `11.0.22000.160` . But I also noticed that subtree I expanded would turn into correct (binded) state (maybe temporarily). I wonder if that was the tree view replaced the calculated state (using the state of sons of this node) with binded state temporarily.
So I'm not sure whether the reason is that my data source is invalid for a tree dependency. Because I think the broken behind logic of a common control shouldn't be still not fixed.
